registry/storage/manifeststore.go

```diff
@@ -154,7 +154,44 @@ func (ms *manifestStore) Put(ctx context.Context, manifest distribution.Manifest
 // Delete removes the revision of the specified manifest.
 func (ms *manifestStore) Delete(ctx context.Context, dgst digest.Digest) error {
 	dcontext.GetLogger(ms.ctx).Debug("(*manifestStore).Delete")
-	return ms.blobStore.Delete(ctx, dgst)
+
+	if !ms.blobStore.deleteEnabled {
+		return distribution.ErrUnsupported
+	}
+
+	// Ensure the blob is available for deletion
+	_, err := ms.blobStore.Stat(ctx, dgst)
+	if err != nil {
+		return err
+	}
```

Comment:
Do we still need this part as we call `ms.Get()` anyway?

Reply:
We have to have this line to keep unit tests working. There's a unit test that does delete multiple times. If we remove this line the test would fail, as you ms.Get() something that has been deleted and the error messages are formatted differently.

Reply:
I see that those logic comes from `ms.blobStore.Delete(ctx, dgst)`.